packages/notifier/src/notifier.js

```diff
@@ -192,9 +193,8 @@ export const makeNotifierFromAsyncIterable = asyncIterable => {
 
       // otherwise return a promise for the next state.
       if (!optNextPromise) {
-        const nextIterResultP = iterator.next();
-        optNextPromise = E.when(
-          nextIterResultP,
+        const nextIterResultP = E(iteratorP).next();
+        optNextPromise = nextIterResultP.then(
```

Comment:
While it is true that this `.then` presents no hazard compared to the original `E.when`, I see two reasons three reasons to keep the old `E.when`:
   * Much programming is done by cargo culting. Someone reading this code and seeing the `.then` may not understand the subtle reasons why this one is safe, and be "inspired" to use a `.then` when it isn't safe. Since they already know `.then`, seeing one here rather than an `E.when` is a missed opportunity to help them unlearn.
   * Refactoring hazard: The delicate reason why this one is safe occurs on the previous line, with no obvious coupling. If we had linting or typing rules that could warn, ..., but we don't.
   * `E.when` accumulates the debugging traces needed for deep stacks. `.then` and `await` do not.